views/cypress/tests/delivery/shared/warning-messages-test.js

```diff
@@ -0,0 +1,445 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 Open Assessment Technologies SA ;
+ */
+
+import { goToNextItem, endTest, goToPreviousItem } from '../../utils/navigation.js';
+import { interactions, expectInteractions, toggleChoice, expectChoiceChecked } from '../../utils/interactions.js';
+
+/**
+ * Check that item is loaded (= navigation was finished successfully)
+ * Use item interactions and section title to identify item
+ * @param {string} sectionTitle
+ * @param {string} interactionType
+ */
+function expectItemLoaded(sectionTitle, interactionType) {
+    cy.get('[data-control="qti-test-position"]').contains(`Section ${sectionTitle}`);
+    if (interactionType === 'choice') {
+        expectInteractions('choiceInteraction', 2);
+    } else {
+        expectInteractions('textEntryInteraction', 1);
+    }
+}
+
+function answerInteraction(interactionType) {
+    if (interactionType === 'choice') {
+        toggleChoice(0, 1);
+        expectChoiceChecked(0, 1, true);
+    } else {
+        cy.get(interactions.textEntryInteraction).type('This is a text').should('have.value', 'This is a text');
+    }
+}
+
+/**
+ * Check dialog is shown
+ * @param {string} title
+ * @param {string} text
+ * @param {Array<string>} buttons
+ */
+function expectDialog(title, text, buttons) {
+    cy.get('.modal')
+        .should('be.visible')
+        .within(() => {
+            cy.get('.message').should('have.text', `${title}${text}`);
+            cy.get('.buttons button').should('have.length', buttons.length);
+            buttons.forEach(btnText => {
+                cy.get('.buttons button').contains(regexEqualCaseInsensitive(btnText));
+            });
+        });
+}
+
+function expectDialogClosed() {
+    cy.get('.modal').should('not.exist');
+}
+
+function clickDialogButton(btnText) {
+    cy.get('.modal .buttons button').contains(regexEqualCaseInsensitive(btnText)).click();
```

Comment:
Wouldn't it be more correct to use the attribute `[data-controls="ok|cancel"]` to select which button to click?
Or is there some reason why you chose the button text?

Reply:
You are right, but this time I think text is better because I check for button text with `expectDialog` function. So after `expectDialog` assert passes, it's guaranteed that button with such text exists. Seems safer to use same 'button selector' for both `expectDialog` and `clickDialogButton`.

Reply:
Ok for me.